src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -241,6 +241,10 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                         intervalTracker,
                         token);
 
+                    await _packageRestoreManager.RaiseAssetsFileMissingEventForSolutionAsync(
```

Comment:
The check in here isn't free, and it'd be happening during *every* restore. I'd love for us to figure out whether we can do something more efficient here.

Reply:
In addition, the only reason I can think of that an assets file would be missing after a restore is if the restore failed, in which case the customer clicking a button to restore again without any changes to project files is unlikely to succeed.

Reply:
That's valid. 
From an impl perspective, I think the check is here in order to remove the banner when done.